File: MenuServer/MenuServer/Models/Menu.cs
using System.ComponentModel.DataAnnotations;

namespace MenuServer.Models
{
    public class Menu
    {
        [Key]
        public int MenuId { get; set; }

        [Required]
        public string Username { get; set; }

        [Required]
        public int TimeOfDayId { get; set; }
        public TimeOfDay TimeOfDay { get; set; }

        [Required]
        public int DayOfWeekId { get; set; }
        public DayOfWeek DayOfWeek { get; set; }

        [Required]
        public int DishId { get; set; }
        public Dish Dish { get; set; }

    }
}

File: menu-client/src/app/Models/menu.ts
import { Dish } from './dish';

export class Menu {
  menuId: number = 0;
  timeOfDayId: number = 0;
  timeOfDayName: string = '';
  dayOfWeekId: number = 0;
  dayOfWeekName: string = '';
  dishId: number = 0;
  dish: Dish = new Dish();
}
